packages/cli/src/util/dev/server.ts

```diff
@@ -135,9 +135,18 @@ export default class DevServer {
   public proxy: httpProxy;
   public envConfigs: EnvConfigs;
   public files: BuilderInputs;
-  public address: string;
-  public devCacheDir: string;
 
+  private address: URL | undefined;
+  public get addressUrl(): URL {
+    if (!this.address) {
+      throw new Error(
+        'Invalid access to `addressUrl` because `start` has not yet populated `this.address`.'
+      );
+    }
+    return this.address;
+  }
```

Comment:
I wanted to have only the property:

```
private address: URL;
```

but we can't because this property isn't definitely set in the constructor. It's set in `start`.

If we leave it as:

```
private address: URL | undefined;
```

then we have to check for `undefined` before using `this.address`, which is annoying.

I think this end result works best, but definitely open to other options.

Reply:
You can define it as `private address!: URL;` to make it always be defined.

Reply:
You can also assign an initial value like it was doing before one line 178

```
this.addressURL = new URL('http://localhost');
```

Reply:
I try to avoid the non-null assertion if I can. It is more brittle than the other approaches.

I'm afraid an initial value would cause confusing errors if it wasn't overwritten with the proper value. The current code in this PR will throw an error in that case.

Reply:
In that case maybe we should call the possible undefined on `_address` so we don't accidentally try to use it

Reply:
Styfle and I chatted about this some more. We decided that if we are going to add another variable to this giant class in this giant file, we should make it very clearly related to the getter.

Updated: [`18783d9` (#8538)](https://github.com/vercel/vercel/pull/8538/commits/18783d9bbc0feebec68cf34800fb5e60ade4e596)